app/src/main/java/org/wikipedia/language/LanguageUtil.kt

```diff
@@ -0,0 +1,142 @@
+                            langTagList.add(langTag)
+                        }
+                        // A Pinyin keyboard will report itself as zh-CN (simplified), but we want to add
+                        // both Simplified and Traditional in that case.
+                        if (langTag.lowercase(Locale.getDefault()) == AppLanguageLookUpTable.CHINESE_CN_LANGUAGE_CODE && !langTagList.contains(
+                                "zh-TW")) {
+                            langTagList.add("zh-TW")
+                        }
+                    }
+                }
+            }
+            if (langTagList.isNotEmpty()) {
+                localeList = LocaleListCompat.forLanguageTags(StringUtil.listToCsv(langTagList))
+                for (i in 0 until localeList.size()) {
+                    val langCode = localeToWikiLanguageCode(localeList[i])
+                    if (langCode.isNotEmpty() && !languages.contains(langCode) && langCode != "und") {
+                        languages.add(langCode)
+                    }
+                }
+            }
+            return languages
+        }
+
+    @JvmStatic
+    fun localeToWikiLanguageCode(locale: Locale): String {
+        // Convert deprecated language codes to modern ones.
+        // See https://developer.android.com/reference/java/util/Locale.html
+        return when (locale.language) {
+            "iw" -> "he" // Hebrew
+            "in" -> "id" // Indonesian
+            "ji" -> "yi" // Yiddish
+            "yue" -> AppLanguageLookUpTable.CHINESE_YUE_LANGUAGE_CODE
+            "zh" -> chineseLanguageCodeToWikiLanguageCode(locale)
+            else -> locale.language
+        }
+    }
+
+    private fun chineseLanguageCodeToWikiLanguageCode(locale: Locale): String {
+        when (locale.script) {
+            "Hans" -> return AppLanguageLookUpTable.SIMPLIFIED_CHINESE_LANGUAGE_CODE
+            "Hant" -> return AppLanguageLookUpTable.TRADITIONAL_CHINESE_LANGUAGE_CODE
+        }
+
+        // Guess based on country. If the guess is incorrect, the user must explicitly choose the
+        // dialect in the app settings.
+        return if (isTraditionalChinesePredominantInCountry(locale.country)) AppLanguageLookUpTable.TRADITIONAL_CHINESE_LANGUAGE_CODE
+        else AppLanguageLookUpTable.SIMPLIFIED_CHINESE_LANGUAGE_CODE
+    }
+
+    private fun isTraditionalChinesePredominantInCountry(country: String?): Boolean {
+        return TRADITIONAL_CHINESE_COUNTRY_CODES.contains(country)
+    }
+
+    @JvmStatic
+    val firstSelectedChineseVariant: String
+        get() {
+            var firstSelectedChineseLangCode: String? = null
+            for (langCode in WikipediaApp.getInstance().language().appLanguageCodes) {
+                if (langCode.startsWith(AppLanguageLookUpTable.CHINESE_LANGUAGE_CODE)) {
+                    firstSelectedChineseLangCode = langCode
+                    break
+                }
+            }
```

Comment:
Can we use `WikipediaApp.getInstance().language().appLanguageCodes.firstOrNull(...)` here?